### PR TITLE
OCPBUGS-33493: Enable leader election for baremetal operator

### PR DESCRIPTION
**Pull request by honza**
This requires:

* ~https://github.com/openshift/baremetal-operator/pull/353~
* https://github.com/openshift/cluster-baremetal-operator/pull/416

Opening this to try it out in CI

We also revert the provisioning CR pause.

### CODE REVIEW

**Comment by honza** [comment]
/payload-job-with-prs periodic-ci-openshift-release-master-nightly-4.16-e2e-metal-ipi-ovn-ipv6 openshift/cluster-baremetal-operator#416 openshift/baremetal-operator#353

File: data/data/bootstrap/baremetal/files/etc/containers/systemd/metal3-baremetal-operator.container [hunk 1]
```diff
@@ -8,7 +8,7 @@ After=network-online.target ironic-volume.service build-ironic-env.service image
 [Container]
 ContainerName=baremetal-operator
 Image=$METAL3_BAREMETAL_OPERATOR_IMAGE
-Exec=/baremetal-operator -build-preprov-image -webhook-port 0
+Exec=/baremetal-operator -build-preprov-image -webhook-port 0 -enable-leader-election -namespace openshift-machine-api -leader-election-release-on-cancel
```
Comment (zaneb) [inline data/data/bootstrap/baremetal/files/etc/containers/systemd/metal3-baremetal-operator.container:11]: where is `-leader-election-release-on-cancel defined`? it's not near https://github.com/openshift/baremetal-operator/blob/master/main.go#L139-L141

Comment (zaneb) [reply data/data/bootstrap/baremetal/files/etc/containers/systemd/metal3-baremetal-operator.container:11]: Ah, it's in https://github.com/openshift/baremetal-operator/pull/353

**Comment by zaneb** [comment]
/payload-job-with-prs periodic-ci-openshift-release-master-nightly-4.16-e2e-metal-ipi-ovn-dualstack https://github.com/openshift/cluster-baremetal-operator/pull/416 https://github.com/openshift/baremetal-operator/pull/353

**Comment by zaneb** [comment]
/payload-job-with-prs periodic-ci-openshift-release-master-nightly-4.16-e2e-metal-ipi-ovn-dualstack openshift/cluster-baremetal-operator#416 openshift/baremetal-operator#353

**Comment by honza** [comment]
In the output of `metal-ipi-ovn-dualstack`, it looks like control plane provisioning failed.

```
level=debug msg=    baremetalhost resource not yet available, will retry
level=info msg=  baremetalhost: ostest-master-0: uninitialized
level=info msg=  baremetalhost: ostest-master-1: uninitialized
level=info msg=  baremetalhost: ostest-master-2: uninitialized
level=debug msg=Fetching Bootstrap SSH Key Pair...
```

**Comment by honza** [comment]
Running this without

```
Environment="LEASE_DURATION=137s"
Environment="RENEW_DURATION=107s"
Environment="RETRY_PERIOD=26s"
```

seems to work.  Trying to figure out why.

**Comment by honza** [comment]
It should be `RENEW_DEADLINE` and not `RENEW_DURATION`.

**Comment by honza** [comment]
/payload-job-with-prs periodic-ci-openshift-release-master-nightly-4.16-e2e-metal-ipi-ovn-ipv6 openshift/cluster-baremetal-operator#416 openshift/baremetal-operator#353

**Comment by zaneb** [comment]
/hold
Need to wait for openshift/cluster-baremetal-operator#416 to merge first
/approve

**Comment by zaneb** [comment]
/retitle OCPBUGS-33493: Enable leader election for baremetal operator

**Comment by honza** [comment]
/unhold

https://github.com/openshift/cluster-baremetal-operator/pull/416 has merged

/retest-required

**Comment by honza** [comment]
/label acknowledge-critical-fixes-only

This blocks agent installs.

**Comment by danmanor** [comment]
@honza `ci/prow/okd-images` is the only required job that didn't passed yet, can we override this one ?

**Pull request merged**